flink-runtime/src/main/java/org/apache/flink/runtime/rpc/RpcEndpoint.java

```diff
@@ -137,18 +153,26 @@ public final void start() {
 	 * @throws Exception indicating that the rpc endpoint could not be started. If an exception occurs,
 	 * then the rpc endpoint will automatically terminate.
 	 */
-	public void onStart() throws Exception {}
+	public final void doStart() throws Exception {
```

Comment:
Regarding the changes in "AkkaRpcActor.java", why not merge `doStart` into `RpcEndpoint#onStart`? We confuse further reader(maybe ourselves) that we have `start` `doStart` and `onStart`.

Reply:
`doStart` is supposed to be called only from `AkkaRpcActor` and never overriden because it does internal staff like modifying the state.
`onStart` is an abstract method to override and do endpoint specific actions.
This way we make sure that the state change is not overriden and not lost.

Reply:
How about inline state transition in `AkkaRpcActor`? If only we can reduce these `start`s.

Reply:
Well, the state transition belongs to the `RpcEndpoint`, not `AkkaRpcActor`.

Reply:
Make sense.